tobias/tools/submerge.py

```diff
@@ -0,0 +1,164 @@
+import glob
+import os
+import argparse
+import sys
+import pandas as pd
+import subprocess
+from tobias.parsers import add_submerge_arguments
+from tobias.utils.logger import TobiasLogger
+
+
+def run_submerge(args):
+
+    logger = TobiasLogger("SubMerge", args.verbosity)
+    logger.begin()
+    parser = add_submerge_arguments(argparse.ArgumentParser())
+    logger.arguments_overview(parser, args)
+
+    outdir = os.path.dirname(args.output)
+
+    # Get all TFBS files
+    tfbs_files = glob.glob(os.path.join(args.tfbs, "*/*_overview.txt"))
+
+    # Read regions
+    logger.debug("Reading regions from: " + args.regions)
+    checked_regions = []
+    with open(args.regions) as f:
+        for i, line in enumerate(f):
+            splitline = line.strip().split("\t")
+
+            # if more than 6 columns, remove the last columns
+            if len(splitline) > 6:
+                splitline = splitline[:6]
+
+            # check if all necessary columns are present
+            while len(splitline) < 6:
+                match len(splitline):
+                    case 3:
+                        splitline.append(f'region_{i}')  # add a unique identifier
+                    case 4:
+                        splitline.append('.')            # add a score placeholder
+                    case 5:
+                        splitline.append('.')            # add a strand placeholder
+
+            # if strand was not specified, intersect with both
+            if splitline[5] != '+' and splitline[5] != '-':
+
+                splitline[5] = '+'
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+                splitline[5] = '-'
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+            else:
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+    # write checked regions to file
+    logger.debug("Writing checked regions to file")
+    regions = os.path.join(outdir, "regions.tmp")
+    with open(regions, "w") as f:
+        for line in checked_regions:
+            f.write(line + "\n")
+
+    # Read TFs
+    if args.TF is not None:
+        logger.debug("Reading TFs from: " + args.TF)
+        tfs = pd.read_csv(args.TF, header=None, names=["TF"])
+        tfs = tfs["TF"].tolist()
+
+        # get only those TFs that were provided
+        tfbs_files = [tfbs for tfbs in tfbs_files if os.path.basename(os.path.dirname(tfbs)) in tfs]
+
+    # intersect each TFBS file with the regions
+    logger.debug('Intersecting query with all TBFS files')
+    all_intersections = ''
+    for file in tfbs_files:
+        tf = os.path.basename(os.path.dirname(file))
+
+        # remove header
+        headless_file = os.path.join(outdir, f"{tf}.headless")
+        command = f'sed "1d" {file} > {headless_file}'
+        subprocess.run(command, shell=True)
```

Comment:
I think it is possible to load the `*_overview.txt` using pybedtools.BedTool and a file handle. With the file handle it should be possible to skip the header line.